src/handlers.rs

```diff
@@ -994,3 +1023,124 @@ fn handle_truncate(
     Ok(empty_buf())
   })
 }
+
+fn handle_listen(
+  state: Arc<IsolateState>,
+  base: &msg::Base,
+  data: &'static mut [u8],
+) -> Box<Op> {
+  assert_eq!(data.len(), 0);
+  if !state.flags.allow_net {
+    return odd_future(permission_denied());
+  }
+
+  let cmd_id = base.cmd_id();
+  let msg = base.msg_as_listen().unwrap();
+  let network = msg.network().unwrap();
+  assert_eq!(network, "tcp");
+  let address = msg.address().unwrap();
+
+  Box::new(futures::future::result((move || {
+    // TODO properly parse addr
```

Comment:
What's not proper about it?

Reply:
It doesn't handle port only: ":8080"
needs tests to ensure conformance with golang.